FirebaseFunctionsSwift/Tests/IntegrationTests.swift

```diff
@@ -0,0 +1,652 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+import FirebaseFunctions
+import FirebaseFunctionsSwift
+import FirebaseFunctionsTestingSupport
+import XCTest
+
+/// This file was intitialized as a direct port of the Objective C
+/// FirebaseFunctions/Tests/Integration/FIRIntegrationTests.m
+///
+/// The tests require the emulator to be running with `FirebaseFunctions/Backend/start.sh synchronous`
+/// The Firebase Functions called in the tests are implemented in `FirebaseFunctions/Backend/index.js`.
+
+struct DataTestRequest: Encodable {
+  var bool: Bool
+  var int: Int32
+  var long: Int64
+  var string: String
+  var array: [Int32]
+  // NOTE: Auto-synthesized Encodable conformance uses 'encodeIfPresent' to
+  // encode Optional values. To encode Optional.none as null you either need
+  // to write a manual encodable conformance or use a helper like the
+  // propertyWrapper here:
+  @NullEncodable var null: Bool?
```

Comment:
I think this is from another set of tests.. but is this necessary for customers to use the `@NullEncodable` wrapper? i.e. what happens if we don't use it, does Codable fail? I'm guessing it means the key doesn't get written in the backend, but maybe it's fine for Functions that the key doesn't exist? If this is required, maybe we need to publicize this and make it part of the public API. It may be helpful to add the context as to how this relates to the Functions backend as well in the comment.

Reply:
My thoughts is that it was likely added since one of the tests contain examples of each primitive encodable type. 

I'd say that if someone needs to represent nil to their callable function, then they would likely know how to do this by either adding a wrapper like this themselves or implement the `Codable` conformance by hand instead of using the synthesized version.

Reply:
SGTM, thanks! Discussed offline and I agree it shouldn't be part of the API surface. Can always revisit in the future if we get feedback that it's needed.